pyuvdata/uvdata/tests/test_uvdata.py

```diff
@@ -10948,4 +10948,324 @@ def test_eq_allowed_failures_filename_string(bda_test_file, capsys):
+
+    ant1 = np.unique(uv.antenna_numbers)[0]
+    ant2 = np.unique(uv.antenna_numbers)[1]
+    pol = "xx"
+    flags = uv.get_flags(ant1, ant2, pol, squeeze="none", force_copy=True)
+    if future_shapes:
+        flags[:, :, :] = True
+    else:
+        flags[:, :, :, :] = True
+    inds1, inds2, indp = uv._key2inds((ant1, ant2, pol))
+    uv.set_flags(flags, ant1, ant2, pol)
+    flags2 = uv.get_flags(ant1, ant2, pol, squeeze="none")
+
+    assert np.allclose(flags, flags2)
+    assert not np.allclose(uv.flag_array, True)
+    return
+
+
+@pytest.mark.filterwarnings("ignore:The uvw_array does not match the expected values")
+@pytest.mark.parametrize("future_shapes", [True, False])
+def test_set_nsamples(hera_uvh5, future_shapes):
+    """
+    Test setting nsamples for a given baseline.
+    """
+    uv = hera_uvh5
+    if future_shapes:
+        uv.use_future_array_shapes()
+
+    ant1 = np.unique(uv.antenna_numbers)[0]
+    ant2 = np.unique(uv.antenna_numbers)[1]
+    nsamples = uv.get_nsamples(ant1, ant2, squeeze="none", force_copy=True)
+    if future_shapes:
+        nsamples[:, :, :] = np.pi
+    else:
+        nsamples[:, :, :, :] = np.pi
+    inds1, inds2, indp = uv._key2inds((ant1, ant2))
+    uv.set_nsamples(nsamples, ant1, ant2)
+    nsamples2 = uv.get_nsamples(ant1, ant2, squeeze="none")
+
+    assert np.allclose(nsamples, nsamples2)
+    assert not np.allclose(uv.nsample_array, np.pi)
+    return
+
+
+@pytest.mark.filterwarnings("ignore:The uvw_array does not match the expected values")
+@pytest.mark.parametrize("future_shapes", [True, False])
+def test_set_nsamples_polkey(hera_uvh5, future_shapes):
+    """
+    Test setting nsamples for a given baseline with a specific polarization.
+    """
+    uv = hera_uvh5
+    if future_shapes:
+        uv.use_future_array_shapes()
+
+    ant1 = np.unique(uv.antenna_numbers)[0]
+    ant2 = np.unique(uv.antenna_numbers)[1]
+    pol = "xx"
+    nsamples = uv.get_nsamples(ant1, ant2, pol, squeeze="none", force_copy=True)
+    if future_shapes:
+        nsamples[:, :, :] = np.pi
+    else:
+        nsamples[:, :, :, :] = np.pi
+    inds1, inds2, indp = uv._key2inds((ant1, ant2, pol))
+    uv.set_nsamples(nsamples, ant1, ant2, pol)
+    nsamples2 = uv.get_nsamples(ant1, ant2, pol, squeeze="none")
+
+    assert np.allclose(nsamples, nsamples2)
+    assert not np.allclose(uv.nsample_array, np.pi)
+    return
+
+
+@pytest.mark.filterwarnings("ignore:The uvw_array does not match the expected values")
+def test_set_data_bad_key_error(hera_uvh5):
+    """
+    Test an error is raised when a key has too many values.
+    """
+    uv = hera_uvh5
+
+    ant1 = np.unique(uv.antenna_numbers)[0]
+    ant2 = np.unique(uv.antenna_numbers)[1]
+    data = uv.get_data(ant1, ant2, squeeze="none", force_copy=True)
+    with pytest.raises(ValueError) as cm:
+        uv.set_data(data, (ant1, ant2, "xx", "foo"))
+    assert str(cm.value).startswith("no more than 3 key values can be passed")
```

Comment:
you could just use the `match` keyword in the pytest.raises call (here and below).

Reply:
Done in 0341185